### PR TITLE
DOCSP-32358: add information about component compatibility tables

### PR DESCRIPTION
**Pull request by ccho-mongodb**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-ecosystem/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-32358>
Staging:

- [Modifications to existing section](https://docs-mongodbcom-staging.corp.mongodb.com/drivers/docsworker-xlarge/DOCSP-32358-component-compatibility/about-compatibility/#mongodb-compatibility-tables-1)
- [New section
](https://docs-mongodbcom-staging.corp.mongodb.com/drivers/docsworker-xlarge/DOCSP-32358-component-compatibility/about-compatibility/#component-compatibility-tables)

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [ ] Are all the links working?

### CODE REVIEW

File: source/about-compatibility.txt [hunk 1]
```diff
@@ -28,10 +28,13 @@ The check marks (✓) indicate that the driver can access **all the
...
-continue to function properly, but may not be able to access the features
-introduced in the newer server versions. We recommend using the newest
-compatible driver when upgrading your server version.
+The circled asterisks (⊛) indicate that the driver version works with
+that specific version of MongoDB server, but may not be able to access all
+the features.
+
+If you are **upgrading your server version**, check whether your current driver
+version is compatible. We recommend using the newest compatible driver when
+upgrading your server version.
```
Comment (ccho-mongodb) [inline source/about-compatibility.txt:37]: Reworded this to remove redundant info

File: source/about-compatibility.txt [hunk 2]
```diff
@@ -58,3 +61,16 @@ labeled with major release versions of the driver.
...
 compatible with the version of the language.
+
+Component Compatibility Tables
+------------------------------
+
+In the **Component Compatibility** tables, the columns indicate versions of
+the driver, and the rows indicate the component package names.
+
+The table cells indicate which versions of the component packages are
+compatible for the driver version specified in that column.
```
Comment (mongoKart) [inline source/about-compatibility.txt:72]: S: AFAIK, things are generally compatible "with" (as stated in the next paragraph), not compatible "for"

```suggestion
compatible with the driver version specified in that column.
```

**Review by mongoKart — APPROVED** [review]
lgtm + 1 suggestion

File: source/about-compatibility.txt (hunk 2 above)
Comment (ccho-mongodb) [reply source/about-compatibility.txt:72]: Missed in my proofreading, thanks!

**Pull request merged**